app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java

```diff
@@ -59,8 +59,14 @@ public boolean equals(final Object o) {
         }
 
         final Contact that = (Contact) o;
-        return this.name.equals(that.name)
-                && this.keys.equals(that.keys);
+        for (final IKey key: this.keys) {
```

Comment:
The old way was good, see https://stackoverflow.com/questions/1075656/simple-way-to-find-if-two-different-lists-contain-exactly-the-same-elements